core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -80,14 +110,55 @@ class ControllerContext {
     replicaStates.clear()
   }
 
+  def partitionReplicaAssignment(topicPartition: TopicPartition): Seq[Int] = {
```

Comment:
I intentionally kept the name of this method the same to make it easier to review (keep the diff smaller), ditto for the update methods below. Perhaps we could rename them to `partitionReplicas/updatePartitionReplicas` and the new methods to `partitionAssignment/updatePartitionAssignment`